src/tx-format/policy.md

```diff
@@ -0,0 +1,51 @@
+# Policy
+
+```c++
+enum PolicyType : uint8 {
+    GasLimit = 0,
+    WitnessLimit = 1,
+    Maturity = 2,
+}
+```
+
+| name   | type                                                                                  | description    |
+|--------|---------------------------------------------------------------------------------------|----------------|
+| `type` | [PolicyType](#policy)                                                                 | Type of input. |
+| `data` | One of [GasLimit](#gaslimit), [WitnessLimit](#witnesslimit), or [Maturity](#maturity) | Input data.    |
+
+Transaction is invalid if:
+
+- `type > PolicyType.Maturity`
+
+## GasLimit
+
+| name       | type     | description                                         |
+|------------|----------|-----------------------------------------------------|
+| `gasLimit` | `uint64` | Gas limit for transaction (including predicate gas) |
```

Comment:
I think we can exclude predicate gas from the gas limit and make it entirely dedicated to the script execution.

We can treat the predicate gas as part of the minimum fee because it is the exact value that we plan to charge from the user, while the script execution depends on the state of the blockchain and may be different -> affects the maximum fee.